UPGRADE_NOTES.md

```diff
@@ -1,3 +1,212 @@
+## Upgrading from 0.19 to 0.20
+
+- Update `.crystal-version` file to `0.34.0`
+- Upgrade to crystal 0.34.0
+- Upgrade Lucky CLI (homebrew)
+
+```
+brew update
+brew upgrade crystal-lang # Make sure you're up-to-date. Requires 0.34.0
+brew upgrade lucky
+```
+
+- Upgrade Lucky CLI (Linux)
+
+> Remove the existing Lucky binary and follow the Linux
+> instructions in this section
+> https://luckyframework.org/guides/getting-started/installing#on-linux
+
+- Update versions in `shard.yml`
+  - Crystal should be `0.34.0`
+  - Lucky should be `~> 0.20.0`
+  - Authentic should be `~> 0.5.2`
+  - LuckyFlow should be `~> 0.6.2`
+- Run `shards update`
+
+### General updates
+
+- Update: any `link()` where the `to:` option is a `String` to an action class. e.g. `link("Home", to: "/")` -> `link("Home", to: Home::Index)`. If no action class is available, use `a(href: "")` instead.
+- Remove: the `?` from any `needs` using a predicate method. e.g. `needs signed_in? : Bool` -> `needs signed_in : Bool`
+- Update: your local `ENV["PORT"]` to be `ENV["DEV_PORT"]`.
+- Update: all `SaveOperation` classes where a raw hash is being passed in. e.g. `MyOperation.new({"name" => "Gary"})` -> `MyOperation.new(Avram::Params.new({"name" => "Gary"}))`
+- Remove: the `on:` option from `needs` inside every Operation class. e.g. `needs created_by : String, on: :create` -> `needs created_by : String`
+
+
+### Optional updates
+
+- Update: all instance variables called from a `needs` on a page can now just use the method of that name. e.g. `@page` -> `page`
+- Add: `include Lucky::CatchUnpermittedAttribute` to the `class Shared::Field(T)` in `src/components/shared/field.cr`
```

Comment:
```suggestion
- Add: `include Lucky::CatchUnpermittedAttribute` to the `class Shared::Field(T)` in `src/components/shared/field.cr`. This will raise a nicer error if you forget to permit a column in your SaveOperation
```